pkg/clustermesh/remote_cluster.go

```diff
@@ -143,8 +137,8 @@ func (rc *remoteCluster) Remove() {
 	rc.mesh.conf.RemoteIdentityWatcher.RemoveRemoteIdentities(rc.name)
 	rc.mesh.globalServices.OnClusterDelete(rc.name)
 
-	if rc.config != nil {
-		rc.usedIDs.ReleaseClusterID(rc.config.ID)
+	if rc.clusterID != cmtypes.ClusterIDUnset {
```

Comment:
nit: We don't need that check, right? 
you've added test that covers this case.

Reply:
Yeah, makes sense to me. Let's dry it up.